Python/Product/Analysis/LanguageServer/IServer.cs

```diff
@@ -20,12 +20,11 @@
 
 namespace Microsoft.PythonTools.Analysis.LanguageServer {
     public interface IServer {
-        Task<InitializeResult> Initialize(InitializeParams @params);
+        Task<InitializeResult> Initialize(InitializeParams @params, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
FYI, these are not backwards compatible changes. To change function parameters in an interface we need to add a new interface and retain support for the old one as well.

Reply:
Have we published this interface already?

Reply:
Wouldn't API changes be okay since we just bumped the assembly version to 16.0?

Reply:
This doesn't solve the problem with `ServerBase` which has to provide overloads which may break existing extension methods. For example:

```cs
public class C1 {
    public void M1(int i) => Console.WriteLine(i);
}

public static void M1(this C1 c1, int i, int j) => Console.WriteLine(i + j);
```

If for the new version you'll change it to
```cs
public class C1 {
    public void M1(int i) => Console.WriteLine(i);
    public void M1(int i, int j) => Console.WriteLine(i - j);
}
```

Class method will be called instead of extension one. This requires recompilation, but after that it will provide quiet semantics change.